conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -80,6 +87,90 @@ public void handlesServiceException() throws IOException {
                 .contains("\"parameters\":{\"foo\":\"bar\"}}");
     }
 
+    @Test
+    public void checkServiceExceptionArgumentsSerializedWithToString() throws IOException {
+        record Argument(
+                String field1, Optional<Integer> maybeField2, List<String> collectionField, String nullString) {}
+        exception = new ServiceException(
+                ErrorType.CONFLICT,
+                SafeArg.of("arg", new Argument("foo", Optional.of(42), List.of("bar", "baz"), null)));
+        HttpURLConnection connection = execute();
+
+        assertThat(connection.getResponseCode()).isEqualTo(ErrorType.CONFLICT.httpErrorCode());
+        String expectedSerializedArg =
+                "\"Argument[field1=foo, maybeField2=Optional[42], collectionField=[bar, baz], nullString=null]\"";
+        assertThat(getErrorBody(connection))
+                .contains("{\"errorCode\":\"CONFLICT\"")
+                .contains("\"parameters\":{\"arg\":" + expectedSerializedArg + "}}");
+    }
+
+    @Test
+    public void handlesCheckedServiceException() throws IOException {
+        final class DifferentPackage extends CheckedServiceException {
+            private DifferentPackage(@Nullable Throwable cause) {
+                super(ErrorType.CONFLICT, cause);
+            }
+        }
+
+        exception = new DifferentPackage(null);
+        HttpURLConnection connection = execute();
+
+        assertThat(connection.getResponseCode()).isEqualTo(ErrorType.CONFLICT.httpErrorCode());
+        assertThat(getErrorBody(connection))
+                .contains("{\"errorCode\":\"CONFLICT\"")
+                .contains("\"parameters\":{}}");
+    }
+
+    @Test
+    public void handlesCheckedServiceExceptionWithOptionals() throws IOException {
+        record Argument(
+                Integer integer,
+                Integer nullValue,
+                OptionalInt optionalInt,
+                OptionalInt emptyOptionalInt,
+                OptionalDouble optionalDouble,
+                OptionalDouble emptyOptionalDouble,
+                Optional<String> optional,
+                Optional<String> emptyOptional,
+                Optional<List<String>> optOfList,
+                List<Optional<String>> listOfOpt) {}
```

Comment:
I don't think this tells us much, we can trust the conjure ObjectMapper to serialize conjure-generated objects correctly (we have plenty of tests for that!)

it's the top-level fields used as SafeArg and UnsafeArg values themselves that are important to test here.